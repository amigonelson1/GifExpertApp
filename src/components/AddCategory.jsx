import { useState } from 'react';

function AddCategory({ /*setCategories*/ onNewCategory }) {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = (e) => {
        setInputValue(e.target.value);
    };

    const onSubmit = (e) => {
        e.preventDefault();
        let valueReturn = inputValue.trim();
        if (valueReturn.length <= 1) return;
        //setCategories(categories => [inputValue, ...categories]);// primero forma de retorna valores al comp padre;
        onNewCategory(valueReturn);// segunda forma de retornar valores al comp padre;
        setInputValue('');
    }

    return (
        <form onSubmit={onSubmit}>
            <input type="text" placeholder="Ingrese la categorÃ­a" value={inputValue}
                onChange={onInputChange}>
            </input>
            <button type="submit">
                Agregar
            </button>
            <button type="reset" onClick={e => setInputValue('')}>
                Limpiar
            </button>
        </form>
    )
}

export default AddCategory