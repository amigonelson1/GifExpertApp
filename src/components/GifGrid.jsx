import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifItem } from './GifItem';

export function GifGrid({ category }) {

    const { images, isLoading } = useFetchGifs(category);

    return (
        <>
            <h3>{category}</h3>
            {isLoading && (<h2>Loading...</h2>)}
            {
                images.length === 0 && (
                    <h4>Lo sentimos, no se encontraron resultados</h4>
                )
            }
            <div className='card-grid'>
                {images.map((images) =>
                    <GifItem key={images.id} {...images} />
                )}
            </div>
        </>
    )
};