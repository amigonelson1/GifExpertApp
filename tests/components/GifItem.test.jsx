import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('pruebas en <GifItem />', () => {
    const title = 'Saitama';
    const url = 'https://saitama.com/';

    test('debe hacer match con snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test('deben ser iguales titulo y url', () => {
        render(<GifItem title={title} url={url} />);
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('debe mostrar el titulo', () => {
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    })
})