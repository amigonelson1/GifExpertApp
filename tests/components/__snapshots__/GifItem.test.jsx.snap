// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifItem /> debe hacer match con snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://saitama.com/"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
